api/routes/otp/controllers.js

```diff
@@ -0,0 +1,37 @@
+const DB = require('models');
+const ApiError = require('base/error');
+const BaseDetailController = require('base/controllers/detailController');
+const { randomGenerator } = require('utils/random');
+
+class OtpLoginController extends BaseDetailController {
```

Comment:
Not a requested change but this is a good use case for a Mixin that injects the getObject method in the controller

Reply:
On a different note, why do we even need a Detail Controller? The basic Model Controller should do the job. We are already overriding `getObject` without using the super and also not utilizing the get handler.